services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -720,14 +719,12 @@ protected void initAttestationTopicSubscriber() {
 
   protected void initActiveValidatorTracker() {
     LOG.debug("BeaconChainController.initActiveValidatorTracker");
+
     final StableSubnetSubscriber stableSubnetSubscriber =
         beaconConfig.p2pConfig().isSubscribeAllSubnetsEnabled()
             ? AllSubnetsSubscriber.create(attestationTopicSubscriber, spec.getNetworkingConfig())
-            : new ValidatorBasedStableSubnetSubscriber(
-                attestationTopicSubscriber,
-                new Random(),
-                spec,
-                beaconConfig.p2pConfig().getMinimumSubnetSubscriptions());
+            : new NodeBasedStableSubnetSubscriber(
```

Comment:
since this is no more dependent on number of active validator, should we have it injected in the active validator tracker? since it is node based, seems to me that should leave independently by the tracker.

Reply:
Makes sense. I separated it from the validator tracker and made it listen to the `SlotEventsChannel`.
@zilm13, this joins your [comment](https://github.com/Consensys/teku/pull/7341#discussion_r1261281048) where you suggested to move it to the `GossipForkSubscriptionsPhase0`. The problem is that we need the `Eth2P2PNetwork` to get the `discoveryNodeId` in order to calculate the subnet ids and the `GossipForkSubscriptionsPhase0` is initialised within the `Eth2P2PNetwork` builder (circular dependency, need the `Eth2P2PNetwork` in order to initialise the `StableSubnetSubscriber`...). I made it completely independant from those. Let me know if that's ok for you or if I'm missing something here

Reply:
Ok, I see your changes.
I think we could definitely subscribe not in validator related block. I will think whether we could move it out of `BeaconChainController` (and should we?)

Reply:
My bad, I moved the subscription to `initSubnetSubscriber`